spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/DeployerSocketUtils.java

```diff
@@ -0,0 +1,296 @@
+ * UDP port refers to a port for a {@link DatagramSocket}.
+ *
+ * <p>
+ *
+ * @author Sam Brannen
+ * @author Ben Hale
+ * @author Arjen Poutsma
+ * @author Gunnar Hillert
+ * @author Gary Russell
+ * @since 4.0
+ */
+public abstract class DeployerSocketUtils {
+
+	/**
+	 * The default minimum value for port ranges used when finding an available socket
+	 * port.
+	 */
+	public static final int PORT_RANGE_MIN = 1024;
+
+	/**
+	 * The default maximum value for port ranges used when finding an available socket
+	 * port.
+	 */
+	public static final int PORT_RANGE_MAX = 65535;
+
+	private static final Random random = new Random(System.nanoTime());
+
+	/**
+	 * Find an available TCP port randomly selected from the range
+	 * [{@value #PORT_RANGE_MIN}, {@value #PORT_RANGE_MAX}].
+	 * @return an available TCP port number
+	 * @throws IllegalStateException if no available port could be found
+	 */
+	public static int findAvailableTcpPort() {
+		return findAvailableTcpPort(PORT_RANGE_MIN);
+	}
+
+	/**
+	 * Find an available TCP port randomly selected from the range [{@code minPort},
+	 * {@value #PORT_RANGE_MAX}].
+	 * @param minPort the minimum port number
+	 * @return an available TCP port number
+	 * @throws IllegalStateException if no available port could be found
+	 */
+	public static int findAvailableTcpPort(int minPort) {
+		return findAvailableTcpPort(minPort, PORT_RANGE_MAX);
+	}
+
+	/**
+	 * Find an available TCP port randomly selected from the range [{@code minPort},
+	 * {@code maxPort}].
+	 * @param minPort the minimum port number
+	 * @param maxPort the maximum port number
+	 * @return an available TCP port number
+	 * @throws IllegalStateException if no available port could be found
+	 */
+	public static int findAvailableTcpPort(int minPort, int maxPort) {
+		return SocketType.TCP.findAvailablePort(minPort, maxPort);
+	}
+
+	/**
+	 * Find the requested number of available TCP ports, each randomly selected from the
+	 * range [{@value #PORT_RANGE_MIN}, {@value #PORT_RANGE_MAX}].
+	 * @param numRequested the number of available ports to find
+	 * @return a sorted set of available TCP port numbers
+	 * @throws IllegalStateException if the requested number of available ports could not
+	 * be found
+	 */
+	public static SortedSet<Integer> findAvailableTcpPorts(int numRequested) {
+		return findAvailableTcpPorts(numRequested, PORT_RANGE_MIN, PORT_RANGE_MAX);
+	}
+
+	/**
+	 * Find the requested number of available TCP ports, each randomly selected from the
+	 * range [{@code minPort}, {@code maxPort}].
+	 * @param numRequested the number of available ports to find
+	 * @param minPort the minimum port number
+	 * @param maxPort the maximum port number
+	 * @return a sorted set of available TCP port numbers
+	 * @throws IllegalStateException if the requested number of available ports could not
+	 * be found
+	 */
+	public static SortedSet<Integer> findAvailableTcpPorts(int numRequested, int minPort, int maxPort) {
```

Comment:
I am in favor of ditching all of the unused API footprint. Lets only keep what is absolutely necessary. Wdyt?

Reply:
Agreed.  I removed all unessential methods and created https://github.com/spring-cloud/spring-cloud-deployer-local/issues/215 to enhance this behavior.